### ISSUE TITLE
Wet gate can't find sample if input has face

### ISSUE THREAD

**Issue by Fang-** [urbit/arvo#1089]
```hoon
> (|*(l=(lest) i.l) ~[1 2])
1
> (|*(l=(lest) i.l) ^-((lest @) ~[1 2]))
1
> (|*(l=(lest) i.l) ^-(foo=(lest @) ~[1 2]))
-find.i.l
> (|*(l=(lest) +<) ^-(foo=(lest @) ~[1 2]))
l=foo=[i=1 t=[i=2 t=~]]
```

#391 seems maybe relevant. I wouldn't expect it to have trouble with this.

**Comment by joshuareagan** [comment]
This is expected behavior -- the superfluous `foo` face of the argument is preserved, for the purposes of a wet gate.  That's why the search for `i.l` fails -- it finds `foo` and then stops searching.

Wet gates are designed for genericity, as a way of implementing type polymorphism for functions.  So you can use the same wet gate at multiple call sites in the code, each of which might pass a different argument type.  Name resolution in the wet gate is relative to the type of the call site argument, *not* the type of the wet gate sample.  And because faces are stored as type information, face information is preserved from the original argument.

This is a principled and type-safe implementation of wet gates.  Perhaps there is a better way to implement wet gates, so that superfluous faces are skipped.  But off the top of my head I can't think of such a solution that doesn't complicate wet gate semantics significantly.

Others may be more creative than I am, however!

**Comment by ohAitch** [comment]
This isn't strictly true: the first example is obviously pulling an i.l out
of the sample l=(lest), not the argument ~[1 2] which has neither. But the
mechanism there is some manner fusing, and in particular it doesn't
*remove* faces; as we can see in the concrete sample type l=foo=[i=1 t=[i=2
t=~]], the sample foo sticks around and prevents access down into interior
axes. (Curious what happens if you use a tweaked argument  lest  with ii
and tt as the faces, I'm guessing we'd see a bunch of i=ii= and t=tt=
stacking?)

**Comment by joshuareagan** [comment]
Yes, you're right, you get extra de-referencing from the wet gate sample faces, but you never get old faces thrown away.
